numpy/lib/_function_base_impl.py

```diff
@@ -4400,139 +4267,82 @@ def quantile(a,
     estimation of the underlying population counterparts, represented by the
     unknown :math:`F`, given a data vector ``a`` of length ``n``.
 
-    One type of estimators arises when one considers :math:`F` as the empirical
-    distribution function of the data, i.e.
-    :math:`F(y) = \\frac{1}{n} \\sum_i 1_{a_i \\leq y}`.
-    Then, different methods correspond to different choices of :math:`x` that
-    fulfill the above inequalities. Methods that follow this approach are
-    ``inverted_cdf`` and ``averaged_inverted_cdf``.
-
-    A more general way to define sample quantile estimators is as follows.
-    The empirical q-quantile of ``a`` is the ``n * q``-th value of the
-    way from the minimum to the maximum in a sorted copy of ``a``. The values
-    and distances of the two nearest neighbors as well as the `method`
-    parameter will determine the quantile if the normalized ranking does not
-    match the location of ``n * q`` exactly. This function is the same as
-    the median if ``q=0.5``, the same as the minimum if ``q=0.0`` and the same
-    as the maximum if ``q=1.0``.
-
-    The optional `method` parameter specifies the method to use when the
-    desired quantile lies between two indexes ``i`` and ``j = i + 1``.
-    In that case, we first determine ``i + g``, a virtual index that lies
-    between ``i`` and ``j``, where  ``i`` is the floor and ``g`` is the
-    fractional part of the index. The final result is, then, an interpolation
-    of ``a[i]`` and ``a[j]`` based on ``g``. During the computation of ``g``,
-    ``i`` and ``j`` are modified using correction constants ``alpha`` and
-    ``beta`` whose choices depend on the ``method`` used. Finally, note that
-    since Python uses 0-based indexing, the code subtracts another 1 from the
-    index internally.
-
-    The following formula determines the virtual index ``i + g``, the location
-    of the quantile in the sorted sample:
-
-    .. math::
-        i + g = q * ( n - alpha - beta + 1 ) + alpha
+    By default, this is done by interpolating between adjacent elements in
+    ``y``, a sorted copy of `a`::
 
-    The different methods then work as follows
+        (1-g)*y[j] + g*y[j+1]
 
-    inverted_cdf:
-        method 1 of H&F [1]_.
-        This method gives discontinuous results:
+    where the index ``j`` and coefficient ``g`` are the integral and
+    fractional components of ``q * (n-1)``, and ``n`` is the number of
+    elements in the sample.
 
-        * if g > 0 ; then take j
-        * if g = 0 ; then take i
+    This is a special case of Equation 1 of H&F [1]_. More generally,
 
-    averaged_inverted_cdf:
-        method 2 of H&F [1]_.
-        This method gives discontinuous results:
+    - ``j = (q*n + m - 1) // 1``, and
+    - ``g = (q*n + m - 1) % 1``,
 
-        * if g > 0 ; then take j
-        * if g = 0 ; then average between bounds
+    where ``m`` may be defined according to several different conventions.
```

Comment:
To respond to [this](https://github.com/numpy/numpy/pull/25704/files#r1589734258),
```suggestion
    where the ``- 1`` in these formulas accounts for Python's 0-based indexing,
    and ``m`` may be defined according to several different conventions.
```
I suppose I left this out original because I chose not to explain all of the ways in which the formulas are not written as they are in the paper. For instance, I also use `%` to take the fractional part rather than subtracting the integral part. This seems equally notable to me. While adding such information would help those who want to compare this to the original text, I think make it harder to follow for those who are reading only this document.

Reply:
`//` and `%` are clear and don’t need explanation.
The different base of indexing is in my opinion worth mentioning somewhere. It took me a while to figure it out.
Maybe a good place is below the reference.

Reply:
https://github.com/numpy/numpy/pull/25704/files#r1590013796